ExtraDry/ExtraDry.Server/RuleEngine.cs

```diff
@@ -147,17 +147,28 @@ private async Task ProcessIndividualUpdate<T>(RuleAction action, PropertyInfo pr
             }
         }
         else {
-            var same = (result == null && destinationValue == null) || (result?.Equals(destinationValue) ?? false);
+            bool same = AreEqual(result, destinationValue);
             if(action == RuleAction.Block && !same) {
-                if(ignore?.Condition == JsonIgnoreCondition.Always) {
+                if(ignoreAttribute?.Condition == JsonIgnoreCondition.Always) {
                     return;
                 }
                 throw new DryException($"Invalid attempt to change property '{property.Name}'", $"Attempt to change read-only property '{property.Name}'");
             }
+            // Do not allow property to be set to the value configured in DeleteValue.
+            
+            if(!same && rulesAttribute != null && rulesAttribute.HasDeleteValue() && AreEqual(value, rulesAttribute.GetDeleteValue())) {
+                throw new DryException($"Invalid attempt to change property '{property.Name}'", $"Please use the Delete function to update '{property.Name}'");
```

Comment:
Independent of above, this message needs improvement.  The first message is for developers and can have as much technical detail as you like.  The second message, the 'userMessage' is designed as a fallback should this exception ever make it all the way to Blazor.  Hopefully it doesn't, but if it does this message can be put on toast delivered to the end-user.  As such your message is too technical.  Perhaps "Unable to update, an attempt was made to make the entity appear deleted.  Check your values to prevent this or use the dedicated delete functionality if available."

Reply:
Updated the UserMessage